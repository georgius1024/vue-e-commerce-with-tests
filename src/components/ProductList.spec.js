import { describe, it, expect, beforeAll, beforeEach, vi } from "vitest";
import { mount } from "@vue/test-utils";
import { useRouter } from "vue-router";
import component from "@/components/ProductList.vue";

vitest.mock("vue-router", () => ({
  useRouter: vi.fn(() => ({
    push: () => {},
  })),
}));

const push = vi.fn();
useRouter.mockImplementation(() => ({
  push,
}));

describe("Product List", () => {
  const buildWrapper = () => {
    const propsData = {
      products: [
        {
          id: 101,
          title: "product",
          thumbnail: null,
          description: "Lorem ipsum",
          price: 100,
        },
      ],
    };
    return mount(component, {
      propsData,
    });
  };
  it("mounts without error", () => {
    const wrapper = buildWrapper();
    expect(wrapper.exists).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("navigates on product click", async () => {
    const wrapper = buildWrapper();
    await wrapper.find('[data-testid="product-preview-101"').trigger("click");
    expect(push).toHaveBeenCalled();
  });
});
