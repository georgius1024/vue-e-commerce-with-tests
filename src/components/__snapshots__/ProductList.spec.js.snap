// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Product List > mounts without error 1`] = `
"<div class=\\"d-flex flex-row flex-wrap\\">
  <div data-v-2606fbc7=\\"\\" class=\\"card pointer hover-up mb-2 me-2\\" data-testid=\\"product-preview-101\\"><img data-v-2606fbc7=\\"\\" class=\\"card-img-top\\" alt=\\"preview\\">
    <div data-v-2606fbc7=\\"\\" class=\\"card-body\\">
      <h5 data-v-2606fbc7=\\"\\" class=\\"card-title\\">product</h5>
      <p data-v-2606fbc7=\\"\\" class=\\"card-text\\">Lorem ipsum</p><span data-v-2606fbc7=\\"\\" class=\\"float-end badge rounded-pill fs-5 bg-primary mt-3 me-3\\"> $100</span>
    </div>
  </div>
</div>"
`;
